### PR TITLE
Ship Logs

### PR DESCRIPTION
**Pull request by artokun**
Implemets #6 

```js
➜  spacetraders git:(al/ship-logs) ✗ st ships:list | jsome
{
  ships: {
    "97eb3c09-e1df-43b8-b70a-d3fe72bb47d6": {
      logs: [
        {
          timestamp: {
            seconds: "1546296210",
            nanos: 0
          },
          log: "Initialize Ship Logs"
        }
      ],
     ...
  }
}
```

### CODE REVIEW

File: proto/ships.proto [hunk 1]
```diff
@@ -32,6 +32,12 @@ service ShipConditionsService {
   rpc show (ByShipIdentifier) returns (ShipConditions) {}
 }
 
+service ShipLogsService {
+  rpc list (Empty) returns (ShipLogsList) {}
+  rpc create (CreateShipLogs) returns (ShipLogs) {}
```
Comment (thebrubaker) [inline proto/ships.proto]:
```suggestion
  rpc create (CreateShipLogs) returns (ShipLog) {}
```

Returning all logs on a single create request would be expensive, just return the created log

File: proto/ships.proto [hunk 2]
```diff
@@ -156,3 +172,11 @@ message ShipConditionsList {
 message CreateShipConditions {
   string ship_id = 1;
 }
+
+message ShipLogsList {
+  map<string, ShipLogs> ships = 1;
+}
+
+message CreateShipLogs {
```
Comment (thebrubaker) [inline proto/ships.proto]: Is this not up to date? I thought you had to send timestamp and log

File: ship-logs/src/actions/create.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const storage = require('../storage')
+
+module.exports = function create(call, callback) {
+  const { ship_id, log, timestamp } = call.request
+
+  storage.ships[ship_id] = {
+    logs: [
```
Comment (thebrubaker) [inline ship-logs/src/actions/create.js]:
```suggestion
    logs: [
      ...storage.ships[ship_id],
```

File: ship-logs/src/actions/create.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const create = require('./create')
+const storage = require('../storage')
+
+describe('Create action', () => {
+  it('should initialize new logs for a ship', async done => {
+    storage.ships = {}
+
+    const call = {
+      request: {
+        ship_id: 1
```
Comment (thebrubaker) [inline ship-logs/src/actions/create.test.js]:
```suggestion
        ship_id: 1,
        timestamp: {
          seconds: 1000,
        },
        log: 'Test',
```

File: ship-logs/src/actions/show.test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const show = require('./show');
+const storage = require('../storage');
+
+describe('List action', () => {
```
Comment (thebrubaker) [inline ship-logs/src/actions/show.test.js]:
```suggestion
describe('Show action', () => {
```

**Comment by artokun** [comment]
Applied your changes, thanks for the thoroughness

**Pull request merged**